test/extended/util/disruption/disruption.go

```diff
@@ -84,35 +85,29 @@ type TestData struct {
 // test is being executed. Description is used to populate the JUnit suite name, and testname is
 // used to define the overall test that will be run.
 func Run(f *framework.Framework, description, testname string, adapter TestData, invariants []upgrades.Test, fn func()) {
-	testSuite := &junit.TestSuite{Name: description, Package: testname}
-	test := &junit.TestCase{Name: testname, Classname: testname}
-	testSuite.TestCases = append(testSuite.TestCases, test)
+	testSuite := &junitapi.JUnitTestSuite{Name: description}
 	cm := chaosmonkey.New(func() {
 		start := time.Now()
-		defer finalizeTest(start, test, testSuite, f)
+		defer finalizeTest(start, testname, testname, testSuite, f)
 		defer g.GinkgoRecover()
 		fn()
 	})
-	runChaosmonkey(cm, adapter, invariants, testSuite)
+	runChaosmonkey(cm, adapter, invariants, testSuite, testname)
 }
 
 func runChaosmonkey(
 	cm *chaosmonkey.Chaosmonkey,
 	testData TestData,
 	tests []upgrades.Test,
-	testSuite *junit.TestSuite,
+	testSuite *junitapi.JUnitTestSuite,
+	packageName string,
 ) {
 	testFrameworks := createTestFrameworks(tests)
 	for _, t := range tests {
 		displayName := t.Name()
 		if dn, ok := t.(testWithDisplayName); ok {
 			displayName = dn.DisplayName()
 		}
-		testCase := &junit.TestCase{
-			Name:      displayName,
-			Classname: "disruption_tests",
-		}
-		testSuite.TestCases = append(testSuite.TestCases, testCase)
```

Comment:
Before we would implicitly create test cases at the start, and then try to figure out later if we should inject or update them for flakes/failures. I found this confusing and inconsistent with how we do things everywhere I've seen so far, so now the test cases are added *after* we've run the test and know what's going on. In practice you'll now see the pass/fail TestCases for a flake together in the xml, instead of one being appended to the end.